modApiExt/alter.lua

```diff
@@ -481,13 +481,12 @@ local function isSkill(v)
 	return type(v) == "table" and v.GetSkillEffect ~= nil
 end
 
-local function skillIndexingFn(self, key)
-	if key == "GetSkillEffect" and self.__Id then
-		local t = skillIndex[self.__Id] or self
-		return t.__GetSkillEffect
+local function skillIndexingFn(tbl, key)
+	local skill = _G[tbl.id]
```

Comment:
Is `_G` here always the true `_G`, or is it possible for it to be `skillIndex`? I am concerned in a double nested skill this could lead to infinite recursion. I believe the environment for `skillIndexingFunction` is never changed by your other logic, but wanted to double check.

Reply:
Good catch, it did enter infinite recursion, though not due to `_G` being substituted - we called the outermost function with replaced environment, which then called a nested function with original environment, which then asked for the table from original `_G`, which went back to modApiExt's override...

I added a fix for that, but I'm pretty sure it doesn't fix the case where one would use local functions, since we can't replace their environment beforehand, so they will still ask for the weapon from real `_G` - I think the only solution is to replace skills in `_G` with dummy tables that redirect calls to `GetSkillEffect`, but pass through calls to other fields, ie. what we did in an earlier version of this PR.